README.md

```diff
@@ -10,27 +14,31 @@ It's inarguable at this point that there is substantial demand for layers that s
 
-(This section needs more details)
+This would be a confiscatory soft fork against those already using the opcode, but this could potentially be mitigated by restricting its use through a smaller reduction in valid transactions (e.g. adding a restiction on the number of CATs in a script) rather than a complete disabling of the opcode.
 
 ## Changes that upgrade bitcoin security
 
 Pierre Rochard said recently that while he's skeptical of changes to bitcoin, he's supportive of security improvements. One potential security improvement is the addition of quantum resistant addresses in a "QuBit" soft fork. If done right, it's hard to argue against such a change. It's likely quantum computers will reach the point where they can break bitcoin's signature security within 10 years' time. This drastically undercuts Saylor's timeframe, and even a quantum resistant address type will need to eventually be upgraded once more to an address type secured by specialized quantum security hardware.
 
 This kind of inevitable change means ossification in all cases is undeniably a bad thing for bitcoin, and demonstrates that there are indeed cases where changes are warranted.
 
-## MEV
+## Centralizing MEV (MEVil)
+
+MEV is an important concern with its potential for distorting miner incentives in a way that leads to mining pool centralization. However, there are good reasons why it won't happen that way on bitcoin. First, accounts-based chains are most vulnerable to MEV through transaction-ordering attacks. Bitcoin, with its UTXOs and PSBT swaps, can enable dexes that are relatively resistant to MEV. L2s even more so.
 
-MEV is a valid concern with its potential for distorting miner incentives in a way that leads to mining pool centralization. However, ultimately, there are good reasons why it won't happen that way on bitcoin. First, accounts-based chains are most vulnerable to MEV through transaction-ordering attacks. Bitcoin, with its UTXOs and PSBT swaps, can enable dexes that are relatively resistant to MEV. L2s even more so.
+It is nearly certain that AMMs are impossible on bitcoin without a fork to explicitly intended to enable them (i.e. adding first class token supporting opcodes). Even if bitcoin were to add a SNARK verification opcode, each token balance check would require a separate proof circuit and data of at least hundreds of bytes and more likely kilobytes. Without a SNARK verifier, checking any token balance would require full transaction history for the token balance, including SPV proofs, and some fraud proof mechanism. Because any fork (aside from first class tokens) which enables reduced token balance proof size would also enable layered token trading, no material token trading would occur on base layer. Why pay for hundreds of bytes of token proof for each trade when you can potentially do a PSBT swap for less than 100 bytes of signature data and lift your token in to a layer where you can trade for very low fees?
```

Comment:
```suggestion
It is nearly certain that AMMs are impossible on bitcoin without a fork explicitly intended to enable them (i.e. adding first class token supporting opcodes). Even if bitcoin were to add a SNARK verification opcode, each token balance check would require a separate proof circuit and data of at least hundreds of bytes and more likely kilobytes. Without a SNARK verifier, checking any token balance would require full transaction history for the token balance, including SPV proofs, and some fraud proof mechanism. Because any fork (aside from first class tokens) which enables reduced token balance proof size would also enable layered token trading, no material token trading would occur on base layer. Why pay for hundreds of bytes of token proof for each trade when you can potentially do a PSBT swap for less than 100 bytes of signature data and lift your token in to a layer where you can trade for very low fees?
```